### DISCUSSION TITLE
Artifacts/Blur when using inpaint engine.

### DISCUSSION THREAD

**Discussion by BoJangles002** [lllyasviel/Fooocus#3190] [Q&A] [answered]
EDIT: inpaint engine does not work with pony based models. 

As of 6/23 using inpaint engine v26 (or 25) is giving me heavy artifacts and blurred images. I was using it without issue on 6/22 where it was blending seamlessly into the original picture. 
![flowers](https://github.com/lllyasviel/Fooocus/assets/48421024/f0169851-ecd7-4fd4-b35d-c371a2e119da)
Any ideas? Using autismmix DPO for my checkpoint. The issue only arises when using inpaint engine.

**Comment by mashb1t** [comment]
Last update was on Jun 21, 2024. Please make sure to use performance "Speed" or "Quality" and no performance LoRA with any inpaint engine except "None" (used in improve details)

**Reply by BoJangles002** [reply]
It looks like I was using it on 6/21 (not 22) and it was working, but I was using it after the latest commit on that day so I don't know. It did have me download a new checkpoint, change the look of the log, and change what the log records when I ran it on 6/23 though. 
Trying both speed and quality with no luck. No loras. The DPO lora is merged into my model. I'll try the non DPO version later.

**Reply by mashb1t** [reply]
Are you certain you use Fooocus and not a fork of it? Please double-check!

**Reply by BoJangles002** [reply]
I am absolutely certain I am using this repo. I used this link from the main page. 
https://github.com/lllyasviel/Fooocus/releases/download/release/Fooocus_win64_2-1-831.7z 
I installed on 6/21 after the latest commit.

I just tried another checkpoint (no DPO lora) and I'm still getting the blur/artifacts when using inpaint engine.

**Comment by BoJangles002** [comment, marked as answer]
Update: I have found the root issue. Pony based checkpoints do not work with inpaint engine in fooocus. I must have not been using pony when I used it originally.

**Reply by ILikeToasters** [reply]
I have been doing a little testing the last couple days with pony models and I have had some luck with inpainting.  I have used it with mixed results. Sometimes it actually works well and others not so well.

**Reply by BoJangles002** [reply]
I tried 3 pony based models and they all ruin the image when using inpaint engine. If you set it to none, it works fine. 

I have previously used the inpaint engine v26 as a control net in forge and it works great if you set it to end at 35% of the way through image gen. I imagine that will be the solution in fooocus as well if it gets added.

**Reply by ILikeToasters** [reply]
What have you found to be good settings for using the pony models in Fooocus? I have figured out the weird prompting and created a style for it. But still trying to find the best options otherwise.

**Reply by vvytch** [reply]
The inpaint problem only occured to me after a certain unnoticable update that changed the Forced Overwrite of Refiner Switch Step to -1 by default. If you change it to anything above 1 instead, inpaint seems to work with pony models!

**Reply by mashb1t** [reply]
refiner switch value > 1 => deactivate refiner switch, which might be the correct way to work around this issue.

**Reply by BoJangles002** [reply]
> What have you found to be good settings for using the pony models in Fooocus? I have figured out the weird prompting and created a style for it. But still trying to find the best options otherwise.

I missed this reply. Styles are prompt edits right? The "recommended" starter tags are source_anime, score_9, score_8_up, score_7_up, score_6_up, score_5_up, score_4_up. I only use score_9 and source_anime anymore as all the other ones influence the output stylistically but if you're making a style you should probably put them all. Pony is a fried/overtrained model so usually best practice is to lower CFG a tad. I settled on 5 CFG in other frontends but it looks like the 6 value in fooocus worked fine. 



> The inpaint problem only occured to me after a certain unnoticable update that changed the Forced Overwrite of Refiner Switch Step to -1 by default. If you change it to anything above 1 instead, inpaint seems to work with pony models!



> refiner switch value > 1 => deactivate refiner switch, which might be the correct way to work around this issue.

wonderful find!

**Reply by Apestein** [reply]
> I tried 3 pony based models and they all ruin the image when using inpaint engine. If you set it to none, it works fine.
> 
> I have previously used the inpaint engine v26 as a control net in forge and it works great if you set it to end at 35% of the way through image gen. I imagine that will be the solution in fooocus as well if it gets added.

Setting a lower step seems to also work, artifacts was getting added at the end for some reason.
